src/Voron/Data/BTrees/RecentlyFoundTreePages.cs

```diff
@@ -5,145 +5,219 @@
 // -----------------------------------------------------------------------
 
 using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
 using Sparrow.Server;
 
 namespace Voron.Data.BTrees
 {
-    public sealed class RecentlyFoundTreePages
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct FoundTreePageDescriptor
     {
-        public sealed class FoundTreePage : IDisposable
-        {
-            public readonly long Number;
-            public readonly Slice FirstKey;
-            public readonly Slice LastKey;
-            public readonly long[] CursorPath;
-            private ByteStringContext.Scope _firstScope;
-            private ByteStringContext.Scope _lastScope;
+        public const int MaxCursorPath = 10;
 
-            public TreePage Page;
+        public TreePage Page;
+        public long Number;
 
-            public FoundTreePage(long number, TreePage page, Slice firstKey, Slice lastKey, long[] cursorPath, ByteStringContext.Scope firstScope, ByteStringContext.Scope lastScope)
-            {
-                Number = number;
-                Page = page;
-                FirstKey = firstKey;
-                LastKey = lastKey;
-                CursorPath = cursorPath;
-                _firstScope = firstScope;
-                _lastScope = lastScope;
-            }
+        public Slice FirstKey;
+        public ByteStringContext.Scope FirstScope;
 
-            public void Dispose()
-            {
-                _firstScope.Dispose();
-               _lastScope.Dispose();
-            }
-        }
+        public Slice LastKey;
+        public ByteStringContext.Scope LastScope;
 
-        private readonly FoundTreePage[] _cache;
+        // We are setting ourselves to not allow pages whose path sequences are longer than MaxCursorPath, this makes sense
+        // because if we are in such a long sequence, it is highly unlikely this cache would be useful anyway. 
+        public int PathLength;
+        public fixed long PathSequence[MaxCursorPath];
 
-        private readonly int _cacheSize;
+        public ReadOnlySpan<long> Cursor => MemoryMarshal.CreateReadOnlySpan(ref PathSequence[0], PathLength);
+    }
 
-        private int _current = 0;
+    [StructLayout(LayoutKind.Sequential, Pack=1)]
+    public unsafe struct RecentlyFoundTreePages()
+    {
+        // PERF: We are using a cache size that is able to be accessed directly from a 512 bits instruction. 
+        public const int CacheSize = 8;
```

Comment:
Previously we had:

`llt.Flags == TransactionFlags.Read ? 8 : 2`

Is it intentional to have the cache of size 8 for both - read and write transactions?

Reply:
It was a tradeoff, we weren't fast enough back then for write transactions (because the cost of reset was high).